File: Dataset/Leetcode/train/112/705.py
 class Solution(object):
    def XXX(self, root, sum):
        """
        :type root: TreeNode
        :type sum: int
        :rtype: bool
        """
        if not root:
            return False
        if not root.left and not root.right and root.val == sum:
            return True
        else:
            return self.XXX(root.left,sum-root.val) or self.XXX(root.right,sum-root.val)

File: Dataset/Leetcode/test/100/705.java
 class Solution {
    public boolean XXX(TreeNode p, TreeNode q) {
        if(p == null && q == null)return true;
        if(p == null && q != null)return false;
        if(q == null && p != null)return false;
        if(q.val != p.val)return false;
        return XXX(p.left,q.left) && XXX(p.right,q.right);
    }
}

